### DISCUSSION TITLE
<input> and <select> behave differently with reset

### DISCUSSION THREAD

**Discussion by jordanwallwork** [react-hook-form/react-hook-form#3279] [Q&A] [answered]
Hi, I'm having some trouble and am hoping someone can help.

I've got a form which uses a mixture of text inputs and select boxes. I use `formState.isDirty` to detect when the values have been changed and show the submit button. Then once the form is submitted, the data is stored and I call `reset()` which sets `isDirty` back to false.

This works as expected with text inputs, but for some reason sets selects back to their original value (despite the fact that the store now contains the correct value). I've created a sandbox to show this behaviour: https://codesandbox.io/s/react-hook-form-issue-forked-nkre5.

I tried calling `setValue` in the submit handler in case this was what was required but that didn't seem to do anything. Not sure what else to try. I'm sure I'm missing something but can't figure out what. Any help would be greatly appreciated, this has had me stumped for far longer than I care to admit and my head is getting sore from banging it against my desk...

**Comment by bluebill1049** [comment, marked as answer]
hey @jordanwallwork how to reproduce the problem in your csb?

**Reply by jordanwallwork** [reply]
If you change the fields, you'll notice that the watch value changes as expected and the submit button appears. Hitting submit updates the store value, and the text input stays the same, but the select box reverts back to its original value for some reason

**Reply by bluebill1049** [reply]
https://codesandbox.io/s/react-hook-form-issue-forked-wck8f?file=/src/index.js
Take a look at the CSB and i have left some comment

**Reply by jordanwallwork** [reply]
Oh brilliant thanks, using I changed it to use `reset(store);` and it works as expected.

Out of interest though, do you know why the basic `reset()` call maintains the value in text boxes but not in selects? I can't see why the two would work differently

**Reply by bluebill1049** [reply]
I think it's potentially within React, instead of this lib. defaultValue was not meant to be changed/mutated. by invoking reset(xxx), react hook form can mutate the dom node with its value.
